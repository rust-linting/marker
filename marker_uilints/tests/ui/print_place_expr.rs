#[derive(Default)]
struct FieldStruct {
    a: u32,
}

fn main() {
    let mut object = FieldStruct { a: 1 };
    let tuple = (1, 2);
    let array = [1, 2, 3];

    let _print_struct_field = object.a;
    let _print_tuple_field = tuple.0;
    let _print_array_index = array[0];
}
