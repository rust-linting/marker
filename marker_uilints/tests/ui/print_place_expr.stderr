warning: print test
  --> $DIR/print_place_expr.rs:11:5
   |
11 |     let _print_struct_field = object.a;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Field(
               FieldExpr {
                   data: CommonExprData {
                       _lifetime: PhantomData<&()>,
                       id: ExprId(..),
                       span: SpanId(..),
                   },
                   operand: Path(
                       PathExpr {
                           data: CommonExprData {
                               _lifetime: PhantomData<&()>,
                               id: ExprId(..),
                               span: SpanId(..),
                           },
                           path: AstQPath {
                               self_ty: None,
                               path_ty: None,
                               path: AstPath {
                                   segments: [
                                       AstPathSegment {
                                           ident: Ident {
                                               name: "object",
                                               span: $DIR/print_place_expr.rs:11:31 - 11:37,
                                           },
                                           generics: SynGenericArgs {
                                               args: [],
                                           },
                                       },
                                   ],
                               },
                               target: Var(
                                   VarId(..),
                               ),
                           },
                       },
                   ),
                   field: Ident {
                       name: "a",
                       span: $DIR/print_place_expr.rs:11:38 - 11:39,
                   },
               },
           )
   = note: `#[warn(marker::test_lint)]` on by default

warning: print test
  --> $DIR/print_place_expr.rs:12:5
   |
12 |     let _print_tuple_field = tuple.0;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Field(
               FieldExpr {
                   data: CommonExprData {
                       _lifetime: PhantomData<&()>,
                       id: ExprId(..),
                       span: SpanId(..),
                   },
                   operand: Path(
                       PathExpr {
                           data: CommonExprData {
                               _lifetime: PhantomData<&()>,
                               id: ExprId(..),
                               span: SpanId(..),
                           },
                           path: AstQPath {
                               self_ty: None,
                               path_ty: None,
                               path: AstPath {
                                   segments: [
                                       AstPathSegment {
                                           ident: Ident {
                                               name: "tuple",
                                               span: $DIR/print_place_expr.rs:12:30 - 12:35,
                                           },
                                           generics: SynGenericArgs {
                                               args: [],
                                           },
                                       },
                                   ],
                               },
                               target: Var(
                                   VarId(..),
                               ),
                           },
                       },
                   ),
                   field: Ident {
                       name: "0",
                       span: $DIR/print_place_expr.rs:12:36 - 12:37,
                   },
               },
           )

warning: print test
  --> $DIR/print_place_expr.rs:13:5
   |
13 |     let _print_array_index = array[0];
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Index(
               IndexExpr {
                   data: CommonExprData {
                       _lifetime: PhantomData<&()>,
                       id: ExprId(..),
                       span: SpanId(..),
                   },
                   operand: Path(
                       PathExpr {
                           data: CommonExprData {
                               _lifetime: PhantomData<&()>,
                               id: ExprId(..),
                               span: SpanId(..),
                           },
                           path: AstQPath {
                               self_ty: None,
                               path_ty: None,
                               path: AstPath {
                                   segments: [
                                       AstPathSegment {
                                           ident: Ident {
                                               name: "array",
                                               span: $DIR/print_place_expr.rs:13:30 - 13:35,
                                           },
                                           generics: SynGenericArgs {
                                               args: [],
                                           },
                                       },
                                   ],
                               },
                               target: Var(
                                   VarId(..),
                               ),
                           },
                       },
                   ),
                   index: IntLit(
                       IntLitExpr {
                           data: CommonExprData {
                               _lifetime: PhantomData<&()>,
                               id: ExprId(..),
                               span: SpanId(..),
                           },
                           value: 0,
                           suffix: None,
                       },
                   ),
               },
           )

warning: 3 warnings emitted

